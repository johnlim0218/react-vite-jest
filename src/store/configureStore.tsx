import { AnyAction, configureStore, Dispatch, MiddlewareAPI } from '@reduxjs/toolkit';
// import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from '../redux/rootReducer';

const loggerMiddleware = (store: MiddlewareAPI<Dispatch, any>) => (next: Dispatch<AnyAction>) => (action: any) => {
  const result = next(action);
  return result;
};

const middlewares = [loggerMiddleware];

const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => 
    getDefaultMiddleware()
      .concat(
        middlewares
      ),
  devTools: process.env.NODE_ENV !== 'production'
})

export type AppDispatch = typeof store.dispatch;

export default store;
