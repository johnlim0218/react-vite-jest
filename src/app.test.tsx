import * as React from 'react';
import { render, screen } from '@testing-library/react';
import App from './app';
import { createMemoryHistory } from 'history';
import { HashRouter, Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from './store/configureStore';

describe('<App />', () => {
  it('renders component correctly', () => {
    const history = createMemoryHistory();
    history.push('/login');

    render(
      <Provider store={store}>
        <Router location={history.location} navigator={history}>
          <App />
        </Router>
      </Provider>
    );

    const loginTitle = screen.getByText('로그인');
    expect(loginTitle).toBeInTheDocument();

    // const { container } = render(<App />);
    // expect(container.getElementsByTagName('div')).toHaveLength(1);
    // expect(container.getElementsByTagName('div')[0]).toHaveTextContent('테스트');

    // expect(container).toMatchSnapshot();
  });
});
