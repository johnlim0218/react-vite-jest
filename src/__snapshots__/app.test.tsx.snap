// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders component correctly 1`] = `
<div>
  <div>
    테스트
  </div>
</div>
`;
